### PR TITLE
Wallet updates

### PR DESCRIPTION
**Pull request by davidkelly**
Now, just a single wallet file, with the encrypted security image in there too (basically PEM-like format, for now).  While at it, some minor cleanup, and now the wallet file has the username in it, supporting multiple users on a single machine.

### CODE REVIEW

**Comment by davidkelly** [comment]
Of course, your old wallets will not work after this.  So, use the debug endpoint to reset and create a new one.  Note I now return a default pixmap when nothing has been set -- that avoids having some errors in the logs while the backend is getting the salt, and cannot decrypt the image.  You probably will never see it, as that is (should) be 100ms or so, but the logging is annoying.  Similarly, I don't try to decrypt the image until you have a salt, avoiding the error messages for that too.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3230/HighFidelity-Beta-PR11343-187ed71a8ae1fc480bbbba7609619b052b14dc8b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3230/HighFidelity-Beta-PR11343-187ed71a8ae1fc480bbbba7609619b052b14dc8b.dmg

File: interface/src/commerce/Wallet.cpp [hunk 1]
```diff
@@ -39,7 +41,8 @@
 #endif
 
 static const char* KEY_FILE = "hifikey";
-static const char* IMAGE_FILE = "hifi_image"; // eventually this will live in keyfile
+static const char* IMAGE_HEADER = "-----BEGIN SECURITY IMAGE-----\n";
+static const char* IMAGE_FOOTER = "-----END SECURITY IMAGE-----\n";
```
Comment (howard-stearns) [inline interface/src/commerce/Wallet.cpp]: nice. I like the style.

File: interface/src/commerce/Wallet.cpp [hunk 2]
```diff
@@ -357,9 +365,31 @@ bool Wallet::decryptFile(const QString& inputFilePath, unsigned char** outputBuf
...
     }
-    inputFile.open(QIODevice::ReadOnly);
-    QByteArray encryptedBuffer = inputFile.readAll();
+    inputFile.open(QIODevice::ReadOnly | QIODevice::Text);
+    bool foundHeader = false;
+    bool foundFooter = false;
+
+    QByteArray base64EncryptedBuffer;
+
+    while(!inputFile.atEnd()) {
```
Comment (howard-stearns) [inline interface/src/commerce/Wallet.cpp]: coding standard - space between while and (

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3257/HighFidelity-Beta-PR11343-7550374f8f990b822eb4c75c6d7110cf77d546cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3257/HighFidelity-Beta-PR11343-7550374f8f990b822eb4c75c6d7110cf77d546cb.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3264/HighFidelity-Beta-PR11343-bce92a622e266e10ba8d907f93bf87f8511ca724.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3264/HighFidelity-Beta-PR11343-bce92a622e266e10ba8d907f93bf87f8511ca724.dmg

**Pull request merged**